### DISCUSSION TITLE
Help with Magic byte is not 0xE9

### DISCUSSION THREAD

**Discussion by gdreelin** [arendst/Tasmota#11181] [Q&A] [answered]
I have a RBG Bulb that I wanted to upgrade from Tasmota 8.1.0.2 to the current Tasmota v9.3.1 but each time I do I get this "Magic Byte is not 0xE9". I have tried to use the tasmota-minimal.bin.gz file because I was getting "Upload buffer miscompare" when I tried to use tasmota.bin file. It stated online that you have to use the .gz files and start with the minimal file to start with. Now I can't hook the bulb to a serial cable so I have to do this over OTA. How can I get this uploaded or delete the bad bin file and put the new one on?

**Comment by sfromis** [comment, marked as answer]
You should not go directly to current version, but follow the steps detailed in https://tasmota.github.io/docs/Upgrading/ 

I'd suggest using tasmota-lite.bin to get to 8.5.1, and then tasmota-lite.bin.gz to 9.1, and after that current version, where you can upgrade to tasmota.bin.gz. Best to always use the compressed .gz binaries for OTA upgrading after getting past 8.3.1. You'd only need to use tasmota-minimal if your 8.1.0.2 is not tasmota-lite (which it may likely be if it is from Tuya-convert).

**Reply by gdreelin** [reply]
Yeah I tried going from 8.5.1 up but same results. I have not tired the lite version yet it stated to use the minimal one if the buffer is full. I will give that a try thanks

**Reply by gdreelin** [reply]
Oh outstanding that work using the lite version and worked my way up to current. It fixed the corrupt copy too. Thanks for all the help.

**Comment by benziman88** [comment]
Hi, I'm stuck with the same issue! I'm on 8.1 and wanted to update following the upgrade path so beginning with 8.5.1
I tried everything minimal.bin, minimal.bin.gz also after reading this here lite.bin and lite.bin.gz ...
bin.gz gives magic byte is not 0xe9 error and the bin file gives upload buffer error 

Even tried OTA Update which just restarts the device.
Here are some infos about my device not sure where the issue is as I simply bought in on the web with tasmota already isntalled
![grafik](https://user-images.githubusercontent.com/52277532/158981624-65668d3c-aaff-4963-9dab-16244979462f.png)


can anyone help me please?

**Reply by barbudor** [reply]
This is the firmware you should use to go from 8.x to 8.5.1 :  http://ota.tasmota.com/tasmota/release-8.5.1/tasmota-lite.bin
Then from 8.5. 1 to 9.1 : http://ota.tasmota.com/tasmota/release-9.1.0/tasmota-lite.bin.gz
And finally latest current version : http://ota.tasmota.com/tasmota/release/tasmota.bin.gz
If you still get the magic byte error, you have download issues

**Reply by arendst** [reply]
He's crossposting. See https://github.com/arendst/Tasmota/issues/7964#issuecomment-1072273659

**Reply by ak-ha-guy** [reply]
This fixed my problem.  Thank you!

**Comment by benziman88** [comment]
Will remove the other post as I wasn‘t sure if I had the Right thread. Wasn‘t intentional!!
